### PR TITLE
[PR] Feature/issue 354  new feature dynamic relations and roles on cytoscape

### PR DESCRIPTION
**Pull request by lduf**
### Description
Cytoscape refactoring :
- improve perf
- redesign front
- Only use js on cytoscape (no more useless reloading)
- Choose a node type from a dropdown. The node type can only be a list

## Issues fixed
List of related issues that are fixed by this PR :

Fix :
- #354 
- #370 
- #371 
- #374

### Checklist
- [ ] Compiling
- [ ] Code factoring (comments, indent...)
- [ ] Documentation (net4p Doc, Readme...)

### Feature Checklist
- [ ] Cytoscape
- [ ] Person/Index
- [ ] Person/Filter
- [ ] Person/Create
- [ ] Person/Edit
- [ ] Person/Delete
- [ ] Relation/Index
- [ ] Relation/Filter
- [ ] Relation/Create
- [ ] Relation/Edit
- [ ] Relation/Delete
- [ ] Duplicates/Index
- [ ] Duplicates/Filter
- [ ] Duplicates/Create
- [ ] Duplicates/Edit
- [ ] ManageList/Delete
- [ ] ManageList/Create
- [ ] ManageList/Edit
- [ ] ManageField/Index
- [ ] ManageField/Create
- [ ] ManageField/Edit
- [ ] ManageField/Delete
- [ ] ApiList
- [ ] Crews
- [ ] Users

### CODE REVIEW

File: src/resources/js/cytoscape/index.js [hunk 1]
```diff
@@ -25,432 +129,128 @@ function perc2color(perc) {
...
             commands: [ // an array of commands to list in the menu or a function that returns the array
-
                 {
                     content: 'Set as FROM',
                     select: function(ele){
-                        insertParam("from", ele.id())
+                        // set the from selected value
+                        $('#from').val(ele.id());
+                        $('#select2-from-container').attr('title', ele.json()["data"]["name"]).text(ele.json()["data"]["name"]);
+                        view_relative(cy, ele);
```
Comment (mathisfchx) [inline src/resources/js/cytoscape/index.js:245]:
```suggestion
                        setFrom();
```

File: src/resources/js/cytoscape/index.js [hunk 2]
```diff
@@ -611,11 +516,103 @@ function drawGraph(){
...
+        }
 
-}
+        // According to the list, we change the 'type' of each node.
+        // It uses the window.persons variable
+        function updateData(){
+            // foreach nodes
+            cy.nodes().forEach(function(node){
+               //update the data
+                let type = window.persons[node.id()][window.field_list[listNodeTypeId]];
```
Comment (mathisfchx) [inline src/resources/js/cytoscape/index.js:569]: Récupérer plus d'informations au moment de récupérer la list ?

**Review by mathisfchx — CHANGES_REQUESTED** [review]

File: src/resources/js/cytoscape/index.js (hunk 2 above)
Comment (lduf) [reply src/resources/js/cytoscape/index.js]: fix in commit #6cfbd9c1c7afea4ad36d6b36884a9f52fb8adfec

**Review by mathisfchx — APPROVED** [review]
🌍

**Pull request merged**